subprojects/robotpy-wpilib/wpilib/timedrobotpy.py

```diff
@@ -0,0 +1,396 @@
+from sys import argv
+from typing import Any, Callable, Iterable, ClassVar
+from heapq import heappush, heappop, _siftup
+from hal import (
+    report,
+    initializeNotifier,
+    setNotifierName,
+    observeUserProgramStarting,
+    updateNotifierAlarm,
+    waitForNotifierAlarm,
+    stopNotifier,
+    tResourceType,
+    tInstances,
+)
+from wpilib import RobotController
+import wpimath.units
+
+from .iterativerobotpy import IterativeRobotPy
+
+_getFPGATime = RobotController.getFPGATime
+_kResourceType_Framework = tResourceType.kResourceType_Framework
+_kFramework_Timed = tInstances.kFramework_Timed
```

Comment:
This seems like an excessive over-optimisation. These constants are only used once at robot start. Putting this here makes _all_ robot code pay the cost of reading this, even if they're not using `TimedRobot`.